ambari-server/src/main/java/org/apache/ambari/server/api/services/HostComponentService.java

```diff
@@ -35,20 +35,37 @@
 import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.Context;
 import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.UriInfo;
 
 import org.apache.ambari.annotations.ApiIgnore;
 import org.apache.ambari.server.api.resources.ResourceInstance;
 import org.apache.ambari.server.configuration.Configuration;
+import org.apache.ambari.server.controller.HostComponentProcessResponse;
+import org.apache.ambari.server.controller.HostComponentSwagger;
 import org.apache.ambari.server.controller.spi.Resource;
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang.Validate;
 
+import org.apache.http.HttpStatus;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
 /**
  * Service responsible for host_components resource requests.
  */
+@Path("/hosts")
+@Api(value = "Hosts", description = "Endpoint for host-specific operations")
```

Comment:
Actually, this is a nested API. The "/hosts" path is used by HostService. Nested API's don't have @Path annotations (except if they are top level API's at the same time).

The HostComponentService is accessible from HostService.getHostComponentHandler(). That means its path will be synthesized from HostService's top level path ("/hosts") and getHostComponentHandler()'s path ("{hostName}/host_components").

Please see ClusterService (a top level service) and ServiceService (nested under ClusterService) for example.